core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -906,6 +908,25 @@ func (o *evmTxStore) FindReceiptsPendingConfirmation(ctx context.Context, blockN
 	return
 }
 
+// FindTxWithRequestID returns any broadcast ethtx with the given requestID
+func (o *evmTxStore) FindTxWithRequestID(requestID common.Hash) (etx *Tx, err error) {
+	err = o.q.Transaction(func(tx pg.Queryer) error {
+		var dbEtx DbEthTx
+		err = tx.Get(&dbEtx, `SELECT * FROM eth_txes WHERE request_id = $1`, requestID)
+		if err != nil {
+			return pkgerrors.Wrap(err, "FindTxWithRequestID failed to load eth_txes")
+		}
+		etx = new(Tx)
+		DbEthTxToEthTx(dbEtx, etx)
+		err = o.LoadTxAttempts(etx, pg.WithQueryer(tx))
```

Comment:
I don't think you need to make this call. We are just sending the existing Tx to caller, and it doesn't quite require loading the TxAttempts.